sdcm/provision/network_configuration.py

```diff
@@ -89,6 +89,15 @@ def interface_ipv6_address(self):
         else:
             return None
 
+    @property
+    def device(self):
+        # Depend on network configuration:
+        # - "eth0" - if broadcast_address.nic = 0
+        # - "eth1" - if broadcast_address.nic = 1
+        if broadcast_address_config := [conf for conf in self.scylla_network_config if conf["address"] == "broadcast_address"]:
+            return f"eth{broadcast_address_config[0]['nic']}"
```

Comment:
It might be better if we could get the interface names from the node,and hardcode/template it. on some instance types we might get other device names.

Reply:
@fruch 
I get device id from ec2 client. 
```
{'Attachment': 
{'AttachTime': datetime.datetime(2024, 10, 7, 7, 30, 56, tzinfo=tzutc()), 
'AttachmentId': 'eni-attach-0bb1a575eec8d29e5', 
'DeleteOnTermination': True, 
'DeviceIndex': 1, 
'Status': 'attached', 'NetworkCardIndex': 0}, 
'Description': '', 'Groups': [{'GroupName': 'SCT-2-sg', 'GroupId': 'sg-0691c61126d99dd41'}], 'Ipv6Addresses': [{'Ipv6Address': '2a05:d018:12e3:f003:27cd:a725:49cb:939c'}], 
'MacAddress': '0a:e9:43:52:54:3f', 'NetworkInterfaceId': 'eni-0746041ce753fc3bb', 'OwnerId': '797456418907', 'PrivateDnsName': 'ip-10-4-14-86.eu-west-1.compute.internal', 'PrivateIpAddress': '10.4.14.86', 'PrivateIpAddresses': [{'Primary': True, 'PrivateDnsName': 'ip-10-4-14-86.eu-west-1.compute.internal', 'PrivateIpAddress': '10.4.14.86'}], 'SourceDestCheck': True, 'Status': 'in-use', 'SubnetId': 'subnet-09976342f9d62db9f', 'VpcId': 'vpc-05caf4736ea5af013', 'InterfaceType': 'interface'}
```

Or get device name from `/etc/netplan/50-cloud-init.yaml` by `macaddress`.
```
network:
    ethernets:
        ens5:
            dhcp4: true
            dhcp4-overrides: &id001
                route-metric: 100
            dhcp6: true
            dhcp6-overrides: *id001
            match:
                macaddress: 0a:ec:c0:45:cb:dd
            set-name: ens5
        ens6:
            dhcp4: true
            dhcp4-overrides: &id002
                route-metric: 200
            dhcp6: true
            dhcp6-overrides: *id002
            match:
                macaddress: 0a:e9:43:52:54:3f
            set-name: ens6

```

Reply:
changed